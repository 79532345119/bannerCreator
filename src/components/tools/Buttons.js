import React from "react"
import { saveAsPNG, copyHTML } from "../../bll/exports"


export const Buttons = (props) => {

    return (
        <div className = "tools_elem">
            <button className="btn" onClick = {saveAsPNG}>Сохранить в png</button>
            <button className="btn" onClick={e=>copyHTML(props.destLink)}>Скопировать баннер в буфер как html</button>
            <button className="btn" onClick = {props.exportToJSON}>Скопировать конфигурацию как json</button>
        </div>

    )
}