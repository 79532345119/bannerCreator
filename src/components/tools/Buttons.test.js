import React from "react"
import {create} from "react-test-renderer"
import {Buttons} from "./Buttons"


describe("Button component", ()=> {
    test("Button component renders correctly", ()=> {
        const component = create(<Buttons/>);
   
        expect(component).toMatchSnapshot();
    })

})