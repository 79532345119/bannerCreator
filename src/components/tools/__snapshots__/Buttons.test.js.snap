// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Button component renders correctly 1`] = `
<div
  className="tools_elem"
>
  <button
    className="btn"
    onClick={[Function]}
  >
    Сохранить в png
  </button>
  <button
    className="btn"
    onClick={[Function]}
  >
    Скопировать баннер в буфер как html
  </button>
  <button
    className="btn"
  >
    Скопировать конфигурацию как json
  </button>
</div>
`;
